main.go

```diff
@@ -56,22 +63,83 @@ func templateHandler(response http.ResponseWriter, request *http.Request, ps htt
 
 	// Render preseed as default
 	var template string
-	if ps.ByName("template") == "finish" {
```

Comment:
I agree that it's probably time to broaden `templateHandler`, but it might be better to go beyond if/switch and simply allow any template short name to be based in, check for it on disk, and render if it exists.  I.e., /template/preseed would look for preseed.j2, /template/finish look for finish.j2, and /template/mynewthing would look for mynewthing.j2, etc.

And we could add a template_extension config option that defaults to .j2 perhaps.